### ISSUE TITLE
base64: invalid entry

### ISSUE THREAD

**Issue by Itookmyprozac** [justchokingaround/lobster#146]
When I press <Enter> to play the movie it does not work and lobster displays the message "base 64: entrada inválida" (base 64: Invalid entry).

 - OS: Linux Mint 21

**Additional context**
- Lobster installed from source ok.

My config:

```
player=mpv
download_dir=/almacen/descargas/
provider=Vidcloud
subs_language=spanish
debug=1
```

The log file:

```
+ query=
+ [ 1 -gt 0 ]
+ [ Casablanca != Casablanca ]
+ query= Casablanca
+ shift
+ [ 0 -gt 0 ]
+ printf %s  Casablanca
+ tr   -
+ sed s/^-//g
+ query=Casablanca
+ [ 0 = 1 ]
+ [ -z Vidcloud ]
+ [  = 1 ]
+ [  = movie ]
+ [  = tv ]
+ main
+ [ -z  ]
+ [ -z Casablanca ]
+ search Casablanca
+ [ 0 = 1 ]
+ curl -s https://flixhq.to/search/Casablanca
+ sed :a;N;$!ba;s/\n//g;s/class="flw-item"/\n/g
+ sed -nE s@.*<a href=".*/(tv|movie)/watch-.*-([0-9]*)".*title="([^"]*)".*class="fdi-item">([^<]*)</span>.*@\3 (\1) [\4]\t\2@p
+ tee /dev/null
+ response=Casablanca (movie) [1942]	18420
A Night in Casablanca (movie) [1946]	3532
+ [ -z Casablanca (movie) [1942]	18420
A Night in Casablanca (movie) [1946]	3532 ]
+ [ -z Casablanca (movie) [1942]	18420
A Night in Casablanca (movie) [1946]	3532 ]
+ [ 0 = 1 ]
+ [ 0 = 1 ]
+ [ 0 = 0 ]
+ + fzf --reverse --with-nth 1 -d \t --header Choose a Movie or TV Show
printf %s Casablanca (movie) [1942]	18420
A Night in Casablanca (movie) [1946]	3532
+ choice=Casablanca (movie) [1942]	18420
+ printf %s+  Casablanca (movie) [1942]	18420
sed -nE s@(.*) \((movie|tv)\).*@\1@p
+ title=Casablanca
+ printf %s Casablanca (movie) [1942]	18420
+ sed -nE s@(.*) \((movie|tv)\).*@\2@p
+ media_type=movie
+ printf %s Casablanca (movie) [1942]	18420
+ cut -f2
+ media_id=18420
+ [ movie = tv ]
+ keep_running=true
+ loop
+ [ true = true ]
+ get_embed
+ [ movie = movie ]
+ curl -s https://flixhq.to/ajax/movie/episodes/18420
+ tr -d \n
+ sed -nE s_.*href="([^"]*)".*Vidcloud.*_\1_p
+ movie_page=https://flixhq.to/watch-movie/watch-casablanca-18420.2519081
+ printf %s https://flixhq.to/watch-movie/watch-casablanca-18420.2519081
+ sed -nE s_.*-([0-9]*)\.([0-9]*)$_\2_p
+ episode_id=2519081
+ curl -s https://flixhq.to/ajax/sources/2519081
+ sed -nE s_.*"link":"([^"]*)".*_\1_p
+ embed_link=https://rabbitstream.net/embed-4/GdJwg4Sj6uPO?z=
+ [ -z https://rabbitstream.net/embed-4/GdJwg4Sj6uPO?z= ]
+ [ -z https://rabbitstream.net/embed-4/GdJwg4Sj6uPO?z= ]
+ get_json
+ printf %s https://rabbitstream.net/embed-4/GdJwg4Sj6uPO?z=
+ sed -nE s_(.*)/embed-(4|6)/(.*)\?z=$_\1\t\2\t\3_p
+ parse_embed=https://rabbitstream.net	4	GdJwg4Sj6uPO
+ printf %s https://rabbitstream.net	4	GdJwg4Sj6uPO
+ cut -f1
+ provider_link=https://rabbitstream.net
+ printf %s https://rabbitstream.net	4	GdJwg4Sj6uPO
+ cut -f3
+ source_id=GdJwg4Sj6uPO
+ printf %s https://rabbitstream.net	4	GdJwg4Sj6uPO
+ cut -f2
+ embed_type=4
+ curl -s https://rabbitstream.net/ajax/embed-4/getSources?id=GdJwg4Sj6uPO -H X-Requested-With: XMLHttpRequest
+ json_data={"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18}
+ [ -n {"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18} ]
+ extract_from_json
+ json_key=file
+ printf %s {"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18}
+ tr {} \n
+ sed -nE s_.*"file":"([^"]*)".*_\1_p
+ grep \.m3u8
+ encrypted=
+ [ -n  ]
+ json_key=sources
+ printf %s {"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18}
+ tr {} \n
+ sed -nE s_.*"sources":"([^"]*)".*_\1_p
+ encrypted=U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J
+ curl -s https://9anime.eltik.net/key/4
+ key=xK0Vk7IZFoy5XSNredsKM
+ printf %s {"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18}
+ tr {|} \n
+ sed -nE s_.*"sources":"([^"]*)".*_\1_p
+ head -1
+ encrypted_video_link=U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J
+ printf %s U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J
+ base64 -d
+ openssl enc -aes-256-cbc -d -md md5 -k xK0Vk7IZFoy5XSNredsKM
+ sed -nE s_.*"file":"([^"]*)".*_\1_p
base64: entrada inválida
+ video_link=
+ printf %s {"sources":"U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J","tracks":[],"encrypted":true,"server":18}
+ sed -e s|U2FsdGVkX1DBYuB/p/UT9dp4f1vLWdqCKYv12K4I0OexgqmRfPrL0jTymy5lmJqTLRo1UH7w6bAOEsCZLbmfkezOYEhDSKWvgDh+VrqiAgRdu7fqZz30u056oFoDPGFkzp8EZZg3WTb09BsOF56cIUAC6qSJh5TFCb5e4EhPKJiohGjmgu86kSJJwdj03Bq3igc1LmkxVdhafbavqEPVS3zsQJguzIc8Fdch5uHbsxo5JSLcAbsDXVt5n5FlzW+Qxx3UKTowi+OBjDgGUTI6JfHa5oyb6QSIHXv1jPhnwe9QvrhDDkbLvRNHTIoYolR0g4EjdVdCOXCWnTqpvwL7GAOgo7JWH/ae792lqquLoJcr86P3mdBQwWjp1MdWaqU/3OBwrCgIksz16Q11tpzYhYBaAqLwOu0ItHJVfzjW+xOTeAz/qtkt/h8EcvswSHAcAtQtIfhM37Afi+NFttLlMP45sBDt5TZiVQxLj8vVjZPTsKDLEn2fUDA+A7+l3BymHeIuxT2dO29RuvbRydG9aExgZSiY2yfwjhpKJyCG1JPGSGhjpTCsSdlxkun7n2wWO1a/kMtc8nxkiifpzx2oKsHQPVu9KEnW5J||
+ json_data={"sources":"","tracks":[],"encrypted":true,"server":18}
+ [ -n  ]
+ [ 0 = 1 ]
+ printf %s {"sources":"","tracks":[],"encrypted":true,"server":18}
+ tr {} \n
+ sed -nE s@.*"file":"([^"]*)","label":"(.panish)[,"\ ].*@\1@p
+ subs_links=
+ subs_arg=--sub-file
+ printf %s 
+ wc -l
+ num_subs=0
+ [ 0 -gt 0 ]
+ [ -z  ]
+ send_notification No subtitles found
+ [ 0 = 1 ]
+ [ 0 = 0 ]
+ [ -z  ]
+ printf \33[2K\r\033[1;34m%s\n\033[0m No subtitles found
No subtitles found
+ return
+ [ -z  ]
+ exit 1

```

**Comment by justchokingaround** [comment]
tldr: flixhq (or rather rapidstream and dokicloud) changed their way of obfuscating the js, so getting the decryption keys isnt possible atm, working in it

**Comment by justchokingaround** [comment]
closing as duplicate of #144

**Comment by justchokingaround** [comment]
https://github.com/justchokingaround/lobster/commit/dfaf1a3a93f6793127c7cbea20759ef7333a662c should fix this issue

**Comment by Itookmyprozac** [comment]
It did! It's fixed :)

Thanks!
